sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala

```diff
@@ -78,19 +78,23 @@ case class GpuJsonToStructs(
               withResource(isLiteralNull.ifElse(emptyRow, nullsReplaced)) { cleaned =>
                 checkForNewline(cleaned, "\n", "line separator")
                 checkForNewline(cleaned, "\r", "carriage return")
-                // if the last entry in a column is incomplete or invalid, then cuDF
-                // will drop the row rather than replace with null if there is no newline, so we
-                // add a newline here to prevent that
-                val joined = withResource(cudf.Scalar.fromString("\n")) { lineSep =>
-                  cleaned.joinStrings(lineSep, emptyRow)
+
+                // add a newline to each JSON line
+                val withNewline = withResource(cudf.Scalar.fromString("\n")) { lineSep =>
+                  withResource(ColumnVector.fromScalar(lineSep, cleaned.getRowCount.toInt)) {
+                    newLineCol =>
+                      ColumnVector.stringConcatenate(Array[ColumnView](cleaned, newLineCol))
+                  }
                 }
-                val concat = withResource(joined) { _ =>
-                  withResource(ColumnVector.fromStrings("\n")) { newline =>
-                    ColumnVector.stringConcatenate(Array[ColumnView](joined, newline))
+
+                // join all the JSON lines into one string
+                val joined = withResource(withNewline) { _ =>
+                  withResource(Scalar.fromString("")) { emptyString =>
+                    withNewline.joinStrings(emptyString, emptyRow)
```

Comment:
Note that the `naRep` parameter of `emptyRow` here is redundant since we already replaced nulls with `emptyRow` earlier in this code.